celements-spring-security/src/main/java/com/celements/spring/security/oauth2/cookietoken/CompositeBearerTokenResolver.java

```diff
@@ -0,0 +1,32 @@
+package com.celements.spring.security.oauth2.cookietoken;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.server.resource.BearerTokenError;
+import org.springframework.security.oauth2.server.resource.BearerTokenErrors;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenResolver;
+import org.springframework.security.oauth2.server.resource.web.DefaultBearerTokenResolver;
+
+public class CompositeBearerTokenResolver implements BearerTokenResolver {
+
+  private final BearerTokenResolver header = new DefaultBearerTokenResolver();
+  private final BearerTokenResolver cookie;
```

Comment:
naming: if it's explicitly for cookies only, it should be a `CookieBearerTokenResolver cookie`. Otherwise it's just:
```suggestion
  private final BearerTokenResolver defaultResolver = new DefaultBearerTokenResolver();
  private final BearerTokenResolver otherResolver;
```